pfcwd/main.py

```diff
@@ -162,5 +169,34 @@ def stop(ports):
             continue
         configdb.mod_entry("PFC_WD_TABLE", port, None)
 
+# Set WD default configuration on server facing ports when enable flag is on
+@cli.command()
+def start_default():
+    """ Start PFC WD by default configurations  """
+    configdb = swsssdk.ConfigDBConnector()
+    configdb.connect()
+    enable = configdb.get_entry('DEVICE_METADATA', 'localhost').get('default_pfc_wd_status')
+    if enable != "enable":
+       return
+    device_type = configdb.get_entry('DEVICE_METADATA', 'localhost').get('type')
+    if device_type != "ToRRouter":
+        return
+    port_num = len(configdb.get_table('PORT').keys())
+    vlan_members = [p[1] for p in configdb.get_table('VLAN_MEMBER').keys()]
+
+    pfcwd_info = {
+        'detection_time': DEFAULT_DETECTION_TIME * max(port_num/DEFAULT_PORT_NUM, 1),
+        'restoration_time': DEFAULT_RESTORATION_TIME * max(port_num/DEFAULT_PORT_NUM, 1),
+        'action': DEFAULT_ACTION
+    }
+
+    for port in vlan_members:
```

Comment:
using vlan to determine whether the port is connected to server is the last resort, should use neighbor table first. check this commit.

https://github.com/Azure/sonic-buildimage/commit/720c71399c9b4d7d0e5a6083b3f63b887f3f6201

Reply:
I checked the above commit and I believe it also uses vlan to determine the server facing port as we don't have neighbor table for servers.